app/services/foreman_deprecation.rb

```diff
@@ -0,0 +1,24 @@
+class ForemanDeprecation
+  def self.deprecation_warning(moved_to_controller, moved_to_method, foreman_version_deadline, additional_info = nil,callstack = caller)
```

Comment:
moved_to_* seems very specific to controllers and methods, will it be useful for other types of deprecations where whole pieces of functionality have changed?